### PR TITLE
Update setup doc for react native swift

### PR DESCRIPTION
**Pull request by sunnylqm**
## Summary

If you are using swift in react native, the flipper doc has some issues:

1. Missing flags #341 which makes flipper totally ignored so that it can not connect
2. There are 2 wrong lines in `initializeFlipper` function. Need to comment them out to pass the build.

(Maybe) fixes https://github.com/facebook/flipper/issues/948

Also merge changes from https://github.com/facebook/react-native/commit/e5497ca8f6e3b240948fdbeef0ac2a710f25bb56

## Changelog

[doc] Update setup doc for react native swift

## Test Plan

Use flipper for react native without issues

### CODE REVIEW

**Comment by passy** [comment]
@priteshrnandgaonkar Can you take a look?

File: docs/getting-started/react-native-ios.md [hunk 1]
```diff
@@ -151,12 +185,10 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
     #if FB_SONARKIT_ENABLED
       let client = FlipperClient.shared()
       let layoutDescriptorMapper = SKDescriptorMapper(defaults: ())
-      FlipperKitLayoutComponentKitSupport.setUpWith(layoutDescriptorMapper)
```
Comment (priteshrnandgaonkar) [inline docs/getting-started/react-native-ios.md:154]: This should work if you add the pod ` pod 'FlipperKit/FlipperKitLayoutComponentKitSupport', versions['Flipper'], :configuration => 'Debug'`

Regarding

File: docs/getting-started/react-native-ios.md [hunk 2]
```diff
@@ -151,12 +185,10 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
...
       client?.add(FlipperKitLayoutPlugin(rootNode: application, with: layoutDescriptorMapper!))
       client?.add(FKUserDefaultsPlugin(suiteName: nil))
       client?.add(FlipperKitReactPlugin())
       client?.add(FlipperKitNetworkPlugin(networkAdapter: SKIOSNetworkAdapter()))
-      client?.add(FlipperReactPerformancePlugin.sharedInstance())
```
Comment (priteshrnandgaonkar) [inline docs/getting-started/react-native-ios.md:159]: Same for here, you will have to add `pod FlipperReactPerformancePlugin` to make it work.

File: docs/getting-started/react-native-ios.md [hunk 3]
```diff
@@ -43,8 +64,21 @@ def flipper_post_install(installer)
         cflags << '-DFB_SONARKIT_ENABLED=1'
       end
       config.build_settings['OTHER_CFLAGS'] = cflags
+      if (config.build_settings['OTHER_SWIFT_FLAGS'])
```
Comment (priteshrnandgaonkar) [inline docs/getting-started/react-native-ios.md:67]: This change is just relevant for Swift projects, can we have two sections in the docs, one for Objective C and one for Swift.

**Comment by priteshrnandgaonkar** [comment]
@lucasbento, adding you here, as there seems to be swift specific change which will be required in the upgrade helper.

File: docs/getting-started/react-native-ios.md (hunk 3 above)
Comment (sunnylqm) [reply docs/getting-started/react-native-ios.md:67]: Since there is a if condition, should do no harm to OC project right?

Comment (priteshrnandgaonkar) [reply docs/getting-started/react-native-ios.md:67]: Even OC projects have OTHER_SWIFT_FLAGS build settings.

Comment (sunnylqm) [reply docs/getting-started/react-native-ios.md:67]: Even so, still sounds reasonable to use one configuration for all if no actual conflicts right? Imagine one wants to turn a stock rn project into swift, how can he know he should add one more flag without any hint? (there will be no errors at all)

**Comment by lucasbento** [comment]
> @priteshrnandgaonkar: adding you here, as there seems to be swift specific change which will be required in the upgrade helper.

Thank you for the ping!

I assume this will be released on react-native `0.63`, is that correct?

**Comment by priteshrnandgaonkar** [comment]
Yupp, will be worth to add it in a template with a swift specific function.

**Pull request closed without merging**